lighthouse-core/test/gather/gatherers/trace-elements-test.js

```diff
@@ -328,30 +322,73 @@ describe('Trace Elements gatherer - GetTopLayoutShiftElements', () => {
+      .mockResponse('Animation.resolveAnimation', {remoteObject: {objectId: 3}})
       .mockResponse('Runtime.getProperties', {result: [{
         name: 'animationName',
         value: {type: 'string', value: 'beta'},
+      }]});
+    const driver = new Driver(connectionStub);
+    const traceEvents = [
+      makeAnimationTraceEvent('0x363db876c1', 'b', {id: '1', nodeId: 5}),
+      makeAnimationTraceEvent('0x363db876c1', 'n', {compositeFailed: 8192}),
+      makeAnimationTraceEvent('0x363db876c2', 'b', {id: '2', nodeId: 5}),
+      makeAnimationTraceEvent('0x363db876c2', 'n', {compositeFailed: 8192}),
+      makeAnimationTraceEvent('0x363db876c3', 'b', {id: '3', nodeId: 6}),
+      makeAnimationTraceEvent('0x363db876c3', 'n', {compositeFailed: 8192}),
+    ];
+
+    const result = await TraceElementsGatherer.getAnimatedElements({driver}, traceEvents);
+    expect(result).toEqual([
+      {nodeId: 5, animations: [
+        {name: 'alpha', failureReasonsMask: 8192},
+        {failureReasonsMask: 8192},
+      ]},
+      {nodeId: 6, animations: [
+        {name: 'beta', failureReasonsMask: 8192},
+      ]},
+    ]);
+  });
+
+  it('gets animated node ids with composited animations', async () => {
+    const connectionStub = new Connection();
+    connectionStub.sendCommand = createMockSendCommandFn()
+      .mockResponse('Animation.resolveAnimation', {remoteObject: {objectId: 1}})
+      .mockResponse('Runtime.getProperties', {result: [{
+        name: 'animationName',
+        value: {type: 'string', value: 'alpha'},
+      }]})
+      .mockResponse('Animation.resolveAnimation', {remoteObject: {objectId: 2}})
+      .mockResponse('Runtime.getProperties', {result: [{
+        name: 'animationName',
+        value: {type: 'string', value: ''},
       }]})
       .mockResponse('Animation.resolveAnimation', {remoteObject: {objectId: 3}})
       .mockResponse('Runtime.getProperties', {result: [{
         name: 'animationName',
-        value: {type: 'string', value: 'gamma'},
+        value: {type: 'string', value: 'beta'},
       }]});
     const driver = new Driver(connectionStub);
     const traceEvents = [
-      makeAnimationTraceEvent('1', 5),
-      makeAnimationTraceEvent('2', 5),
-      makeAnimationTraceEvent('3', 6),
+      makeAnimationTraceEvent('0x363db876c1', 'b', {id: '1', nodeId: 5}),
+      makeAnimationTraceEvent('0x363db876c1', 'n', {compositeFailed: 0}),
+      makeAnimationTraceEvent('0x363db876c2', 'b', {id: '2', nodeId: 5}),
+      makeAnimationTraceEvent('0x363db876c2', 'n', {compositeFailed: 0}),
+      makeAnimationTraceEvent('0x363db876c3', 'b', {id: '3', nodeId: 6}),
+      makeAnimationTraceEvent('0x363db876c3', 'n', {compositeFailed: 0}),
     ];
 
     const result = await TraceElementsGatherer.getAnimatedElements({driver}, traceEvents);
     expect(result).toEqual([
       {nodeId: 5, animations: [
-        {id: '1', name: 'alpha'},
-        {id: '2', name: 'beta'},
+        {name: 'alpha', failureReasonsMask: 0},
```

Comment:
do we not care about the IDs anymore? curious why we're regressing some of the test coverage here

Reply:
The id was necessary to match animations in `TraceElements` to their compositing failure reasons when we were still reading the trace in the audit.

Since we moved all the trace logic to the trace elements gatherer, having the animation id in `TraceElements` is no longer necessary.

Reply:
Ah, right on 👍 yeah I suppose they have no use anymore :)